README.md

```diff
@@ -27,6 +27,9 @@ let valid = verify("hunter2", &hashed)?;
 
 The cost needs to be an integer between 4 and 31 (see benchmarks to have an idea of the speed for each), the `DEFAULT_COST` is 12.
 
+## Error on truncation
+For broad compatibility, the default set of features allows hashing of passwords longer than 72 bytes (including the null terminator) by truncating the input to the first 72 bytes. However, in certain use cases this can break 2nd pre-image resistance. To enforce strict safety, one should enable the `error_on_truncation` feature. When `error_on_truncation` is enabled, `hash`, `hash_with_result`, `hash_with_salt`, and `verify` will return error for passwords longer than 72 bytes.
+
```

Comment:
I would also mention that _most_ (if not all) libraries in other languages will truncate by default so this option should only be enabled after careful considerations on whether truncation was ever allowed (through this lib or some other languages lib).
Since this library is also exposed in other languages, I'm wondering if a feature is the best way to expose that but I don't have an idea other than exposing some config structs which isn't going to be really used in practice and just clutter the API

Reply:
The proposed change in documentation was incorporated.

Adding a configuration struct is necessarily a breaking change, whereas a feature flag that is disabled by default will not alter the behavior if users upgrade. This is the main reason why I chose to implement it as a feature flag.

Reply:
You can have parallel APIs, e.g. `truncating_hash*` or thereabouts.

A feature does feel like spooky-action-at-a-distance, in that one dependency may activate it while another one expects the non-truncating behavior. Features are supposed to be purely additive.

Reply:
Yeah I was thinking about that after my comment. The right path is probably a different public hash/verify methods that will error if length > 72 and then just call the default hash/verify

Reply:
Or, the `truncating_*` methods can truncate the input if it's too long, which means they'll never hit an error case in the non-truncating versions